students/mattmaeda/stockscreener/exchange/views.py

```diff
@@ -0,0 +1,266 @@
+"""
+Contains the view logic for stockscreener app
+"""
+import datetime
+import iexfinance
+from django.shortcuts import render
+from django.http import HttpResponse
+from django.urls import reverse
+from requests import ConnectionError
+from iexfinance import Stock
+from .models import Screen
+
+# Create your views here.
+def index(request):
+    """ Main page """
+    message ="Pleast enter a stock symbol (e.g., GOOG)"
+    response = None
+
+    if request.method == "POST":
+
+        try:
+            ticker = request.POST.get("ticker", "")
+            symbol = ticker.upper()
+            company = get_stock_info(symbol)
+
+            if company is not None:
+                history = get_price_history(symbol)
+                response = run_screens(history)
+                message = "Here are your results for {}".format(symbol)
+            else:
+                message = "Invalid stock symbol '{}'".format(symbol)
+
+        except ConnectionError as cxn:
+            message = ("Currently experiencing connection problems.\n"
+                       "Please try again later.")
+
+    return render(request, 'index.html', {"message": message,
+                                          "response": response})
+
+
+def get_stock_info(ticker):
+    """ Checks if the ticker is valid
+
+    :param ticker String: the stock ticker
+
+    :return: company name or None if invalid
+    :rtype: String
+
+    """
+    # Let connection errors percolate to the calling function
+    try:
+        stock = Stock(ticker.upper())
+        return stock.get_company().get("companyName")
+
+    except iexfinance.utils.exceptions.IEXSymbolError as exc:
+        return None
+
+    except ValueError as ve:
+        # If empty
+        return None
```

Comment:
You should at least log the exceptions you are catching. And are you sure that you don't want the user to know about them at all?